### PR TITLE
Docs: Release process

### LINKED ISSUE #17651: Document the release process
For starters, explain the current release cycle of the Grafana project.

### PR DESCRIPTION
**Pull request by marefr**
**What this PR does / why we need it**:
Draft documentation of Grafana release process and management. For now the idea is to link to Google doc for the release guide and all the details. 

**Which issue(s) this PR fixes**:
Closes #17651

**Special notes for your reviewer**:

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# Grafana Releases
+
```
Comment (oddlittlebird) [inline RELEASES.md]: Add an intro that tells the reader what this document will tell them. Pretend you are explaining it to a brand new employee.

And as a general practice, avoid stacked headings. Either add an intro or remove one of the headings.

File: RELEASES.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+## Release schedule
+
+Release cadence of first beta-release being cut is 6 weeks.
```
Comment (oddlittlebird) [inline RELEASES.md]: Unclear what "being cut" means.

Also, what happens during those six weeks? @daniellee  told me there was a five-week cadence with one week rest. Is the first week usually planning?

You don't have to write yourself into a strict schedule, just give our imaginary new employee reader an idea of what that six-week release is like so they know what to expect.

File: RELEASES.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+## Branch management and versioning strategy
+
+We use [Semantic Versioning](https://semver.org/).
+
+We maintain a separate branch for each minor release, named `v<major>.<minor>.x`, e.g. v6.0.x, v6.5.x. The release branch is created from the master branch just before  releasing the first beta-release of a minor version, e.g. v6.5.0-beta1.
+
+New features, bug fixes and changes are squashed and merged into the master branch. Bug fixes needed to be included in a patch release are being cherry-picked from the master branch. The master branch is the development branch.
```
Comment (oddlittlebird) [inline RELEASES.md]:
```suggestion
New features, bug fixes, and changes are squashed and merged into the master branch. Bug fixes required in a patch release are being cherry-picked from the master branch. The master branch is the development branch.
```

File: RELEASES.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+## Release management roles
+
```
Comment (oddlittlebird) [inline RELEASES.md]: Again, add some intro text. We want to avoid stacked headings.

File: RELEASES.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+### Release coordinator
+
+Responsible for coordinating entire release series of a minor release, meaning all beta- and patch releases of a minor release.
+
+### Release responible
```
Comment (oddlittlebird) [inline RELEASES.md]:
```suggestion
### Release responsible
```

File: RELEASES.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+We use [Semantic Versioning](https://semver.org/).
+
+We maintain a separate branch for each minor release, named `v<major>.<minor>.x`, e.g. v6.0.x, v6.5.x. The release branch is created from the master branch just before  releasing the first beta-release of a minor version, e.g. v6.5.0-beta1.
+
+New features, bug fixes and changes are squashed and merged into the master branch. Bug fixes needed to be included in a patch release are being cherry-picked from the master branch. The master branch is the development branch.
+
+## Release management roles
+
+### Release coordinator
```
Comment (oddlittlebird) [inline RELEASES.md]: This is confusing. The first heading under Release management roles is an actual role title, but the second one does not look like a role title. Please make consistent.

File: RELEASES.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+Responsible for coordinating entire release series of a minor release, meaning all beta- and patch releases of a minor release.
```
Comment (oddlittlebird) [inline RELEASES.md]:
```suggestion
Person responsible for coordinating the entire release series of a minor release, meaning all beta and patch releases of a minor release.
```

File: RELEASES.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+### Release responible
+
+Assigned by release coordinator or voluntarily requested. Responsible for doing the actual work of an individual release.
```
Comment (oddlittlebird) [inline RELEASES.md]:
```suggestion
Person assigned by the release coordinator or who volunteers. They are responsible for doing the actual work of an individual release.
```

Comment (oddlittlebird) [reply RELEASES.md]: Is this only one person per release?

File: RELEASES.md [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+
+## Create a new milestone
```
Comment (oddlittlebird) [inline RELEASES.md]: It looks like you are starting a description of a process.

Create a new milestone
Pre-release tasks
Check milestone status
and so on

I suggest moving the level 2 headings down one level and collecting them under something that describes the "bucket" they are in, like "Release process."

You could also rewrite it as an ordered list, which would save you from having to add content to the empty heading sectins.

File: RELEASES.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
+**What:** Create a milestone to keep track of every issue and PR to be included in a release
```
Comment (oddlittlebird) [inline RELEASES.md]: Put periods at the end of sentences. I would also change the order to Who, What, When.

File: RELEASES.md [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+
+## Before the release
+
+**When:** At least a day before the target release date
+**What:** Keep track of milestone process to be able to decide/suggest date of release
+**Who:** Release coordinator
+
+### Check status of milestone
+
+Every PR that's included in milestone have proper labeling:
```
Comment (oddlittlebird) [inline RELEASES.md]:
```suggestion
Every PR that's included in a milestone must be properly labeled:
```

File: RELEASES.md [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+- `type/bug` - if the issue it fixes is labeled as a bug
```
Comment (oddlittlebird) [inline RELEASES.md]: Begin each definition with a capital letter, end all sentences with periods.

File: RELEASES.md [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+- `cherry-pick needed` - denoting that a PR needs to be cherry-picked to be included in release. Applicable for any release after first beta release, e.g. release branch exists.
+- `add to changelog` - if the PR should be added to the changelog.
+
+Any PR merged that haven't been added to the release milestone?
```
Comment (oddlittlebird) [inline RELEASES.md]: Why are you asking this sentence here? Unclear.

Do you mean that someone should check to see if there are PRs that have been merged that have not been added to the release milestone?

File: RELEASES.md [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+
+### Decide/suggest date of release
```
Comment (oddlittlebird) [inline RELEASES.md]: Are we deciding or suggesting? And who is doing it?

File: RELEASES.md [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+
+### Assign release responsible
```
Comment (oddlittlebird) [inline RELEASES.md]: Who is assigning them? I know it was mentioned up in the roles section that the release coordinator assigns the responsibility, but you should also state here something like:

The release coordinator assigns the release responsible role.

File: RELEASES.md (hunk 6 above)
Comment (oddlittlebird) [reply RELEASES.md]: Also, I suggest capitalizing the roles so it is clear throughout that these are roles or titles.

Release Coordinator
Release Responsible

File: RELEASES.md [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+
+## Prepare the release
+
+**Preconditions:**
```
Comment (oddlittlebird) [inline RELEASES.md]: If this is presented in an ordered list and is part of the process, then by virtue of where it is, the previous steps should be assumed to have been done. You can probably remove the Preconditions section.

File: RELEASES.md [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+  - Date of release decided
+  - Release responsible assigned
+
+**When:** At least a day before the target release date
+**What:** Prepare everything needed to be ready to draft a new release on the target release date
+**Who:** Release coordinator and release responsible(s)
```
Comment (oddlittlebird) [inline RELEASES.md]:
```suggestion
**Who:** Release coordinator and release responsible
```

Comment (oddlittlebird) [reply RELEASES.md]: "Responsible" does not pluralize well, it also is not a noun. Please rethink using it as a role title. Or at least add "person" or "people" to the end of it.

File: RELEASES.md [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+
+### Update changelog
```
Comment (oddlittlebird) [inline RELEASES.md]: All these empty headings either need content or need to not be headings.

Also, I recommend using sentences to make clear who is doing what here.

File: RELEASES.md [hunk 19]
```diff
@@ -0,0 +1,77 @@
...
+
+### First major/minor beta release?
```
Comment (oddlittlebird) [inline RELEASES.md]: Why are we asking questions? Who is asking the question? What does it mean to me and the process if the answer is one or the other? Unclear.

File: RELEASES.md [hunk 20]
```diff
@@ -0,0 +1,77 @@
...
+
+### Any release after first beta release?
```
Comment (oddlittlebird) [inline RELEASES.md]: See previous comment.

File: RELEASES.md [hunk 21]
```diff
@@ -0,0 +1,77 @@
...
+#### Cherry pick commits
+
+## Tag a new release
+
+**Who:** Release responsible(s)
+**When:** Target release date
+
+See [Grafana Release Guide](https://).
+
+## After the release
```
Comment (oddlittlebird) [inline RELEASES.md]: Ater the release what? After the release is live in production?

**Review by oddlittlebird — COMMENTED** [review]
Great content! This information will be very helpful for new Grafanistas.

File: RELEASES.md (hunk 6 above)
Comment (marefr) [reply RELEASES.md]: Yeah I know. Tried to come up with a good role name of the one doing the actual release. This was my best alternative :) Release manager feels like something else, right?

Do you have a better suggestion?

File: RELEASES.md (hunk 18 above)
Comment (marefr) [reply RELEASES.md]: Yeah sorry should have been more clear that I basically just fleshed out the raw structure now. All headings will have content eventually.

File: RELEASES.md (hunk 6 above)
Comment (oddlittlebird) [reply RELEASES.md]: Release Executor? That sounds all badass. I will see if anyone on Slack has ideas.

**Comment by marefr** [comment]
Planning to pick this up soon

**Comment by marefr** [comment]
Still planning to pick this up

**Comment by marefr** [comment]
Still planning to pick this up, but still too much other things to handle...

**Comment by bergquist** [comment]
@marefr Lets just close this and pick it up once you have time again? :)

Another question I have here is who the audience is?

**Comment by marefr** [comment]
@bergquist idea is to make it more transparent for anyone interested in for example our release schedule (6 weeks). Not a copy of, but inspiration from https://github.com/prometheus/prometheus/blob/master/RELEASE.md.

Instead of closing it I prefer finishing it. I feel I'm soon back at this.

**Pull request closed without merging**